### PR TITLE
added UI for deleteset

### PR DESCRIPTION
**Pull request by Malekai**

### CODE REVIEW

**Comment by JonnyLinja** [comment]
- [ ] Needs to be right aligned
- [ ] Should not have a line above it
- [ ] Should be below REST if it exists

File: app/shared_features/set_card/SetRestRow.js [hunk 1]
```diff
@@ -11,14 +11,17 @@ class SetRestRow extends PureComponent {
 
     render() {
         if (this.props.item.isCollapsed) {
-            var margin = 0;
+            var marginTop = 0;
+            var marginBottom = 15;
```
Comment (JonnyLinja) [inline app/shared_features/set_card/SetRestRow.js]: - [ ] marginBottom needs to be based on if there's a DELETE button as well. If delete button, less margin. IF not delete button, more margin.

File: app/shared_features/set_card/expanded/DeleteSetRow.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import React, {PureComponent} from 'react';
+import {
+    View,
+    StyleSheet,
+    TouchableOpacity,
+    Text
```
Comment (JonnyLinja) [inline app/shared_features/set_card/expanded/DeleteSetRow.js]: - [ ] comma

File: app/shared_features/set_card/SetRestRow.js (hunk 1 above)
Comment (Malekai) [reply app/shared_features/set_card/SetRestRow.js]: A delete button only shows when the item is expanded...

File: app/features/workout/WorkoutScreen.js [hunk 3]
```diff
@@ -81,14 +81,25 @@ const createViewModels = (state, sets) => {
...
         // rest footer
-        if (isInitialSet) {
+        if (isInitialSet && isLastSet) {
+            // new set, reset the end time
+            lastSetEndTime = isRemoved ? null : SetTimeCalculator.endTime(set);
+            array.push(createBottomBorder(set));
+        } else if (isInitialSet && !isLastSet && !isCollapsed) {
             // new set, reset the end time
             lastSetEndTime = isRemoved ? null : SetTimeCalculator.endTime(set);
+            array.push(createDeleteFooter(set));
```
Comment (JonnyLinja) [inline app/features/workout/WorkoutScreen.js]: - [ ] nest the array.push(createDeleteFooter(set)); in the previous if statement

File: app/features/workout/WorkoutScreen.js [hunk 4]
```diff
@@ -81,14 +81,25 @@ const createViewModels = (state, sets) => {
...
-            if (lastSetEndTime !== null) {
+            if (lastSetEndTime !== null && isCollapsed) {
+                array.push(createFooterVM(set, lastSetEndTime, isCollapsed));
+            } else if (lastSetEndTime !== null && !isCollapsed) {
                 array.push(createFooterVM(set, lastSetEndTime, isCollapsed));
+                array.push(createDeleteFooter(set));
+            } else if (!isCollapsed) {
+                array.push(createDeleteFooter(set));
+                array.push(createBottomBorder(set));
             } else {
```
Comment (JonnyLinja) [inline app/features/workout/WorkoutScreen.js]: - [ ] use nested if statements

File: app/features/workout/WorkoutScreen.js [hunk 5]
```diff
@@ -98,6 +109,9 @@ const createViewModels = (state, sets) => {
         } else if (isLastSet && lastSetEndTime !== null && set.reps.length === 0) {
             // working set, live rest mode
             array.push(createWorkingSetFooterVM(set, lastSetEndTime));
+        } else if (!isLastSet && !isCollapsed) {
+            array.push(createDeleteFooter(set));
+            array.push(createBottomBorder(set));
```
Comment (JonnyLinja) [inline app/features/workout/WorkoutScreen.js]: - [ ] use nested if statement

File: app/features/history/HistoryScreen.js [hunk 6]
```diff
@@ -109,20 +109,34 @@ const createViewModels = (state, sets, shouldShowRemoved) => {
         }
 
         // rest footer
-        if (isInitialSet) {
+        if (isInitialSet && isCollapsed) {
```
Comment (JonnyLinja) [inline app/features/history/HistoryScreen.js]: - [ ] same stuff, all of this should be nested if statements

**Comment by JonnyLinja** [comment]
@Malekai Yo, make a PR OFF of this one to make delete set NOT remove the actual data. I can't merge this in right now as current testers would LOSE data if they hit the delete button right now.

**Pull request closed without merging**